craft_parts/packages/rpm.py

```diff
@@ -0,0 +1,200 @@
+            packages = {"mercurial"}
+        elif source_type in ["svn", "subversion"]:
+            packages = {"subversion"}
+        elif source_type == "rpm2cpio":
+            # installed by default in CentOS systems
+            packages = set()
+        elif source_type == "7zip":
+            packages = {"p7zip"}
+        else:
+            packages = set()
+
+        return packages
+
+    @classmethod
+    @functools.lru_cache(maxsize=1)
+    def refresh_packages_list(cls) -> None:
+        """Refresh the list of packages available in the repository."""
+        # Return early when integration testing
+        if os.getenv("CRAFT_PARTS_PACKAGE_REFRESH", "1") == "0":
+            return
+
+        try:
+            cmd = ["yum", "update", "-y"]
+            logger.debug("Executing: %s", cmd)
+            process_run(cmd)
+        except subprocess.CalledProcessError as call_error:
+            raise errors.PackageListRefreshError(
+                "failed to run yum update"
+            ) from call_error
+
+    @classmethod
+    def _check_if_all_packages_installed(cls, package_names: List[str]) -> bool:
+        """Check if all given packages are installed.
+
+        Will check versions if using <pkg_name>=<pkg_version> syntax parsed by
+        get_pkg_name_parts().  Used as an optimization to skip installation
+        and cache refresh if dependencies are already satisfied.
+
+        :return True if _all_ packages are installed (with correct versions).
+
+        XXX: method left out of RPMRepository's MVP; returning False would lead to
+        an inefficiency, this should be implemented in the future.
+        """
+        logger.debug(
+            "Called not implemented method "
+            "'RPMRepository._check_if_all_packages_installed' "
+            "with names %s",
+            package_names,
+        )
+        return False
+
+    @classmethod
+    def download_packages(cls, package_names: List[str]) -> None:
+        """Download the specified packages to the local package cache area.
+
+        XXX: method left out of RPMRepository's MVP; nothing will be
+        downloaded, so raise an error to ensure it's not used.
+        """
+        raise NotImplementedError("Functionality not yet provided by RPMRepository.")
+
+    @classmethod
+    def install_packages(
+        cls,
+        package_names: List[str],
+        *,
+        list_only: bool = False,
+        refresh_package_cache: bool = True,
+    ) -> List[str]:
+        """Install packages on the host system."""
+        if not package_names:
+            return []
+
+        package_names = sorted(package_names)
+        logger.debug("Requested packages: %s", package_names)
+
+        install_required = False
+        if not cls._check_if_all_packages_installed(package_names):
+            install_required = True
+
+        if not list_only:
+            if refresh_package_cache and install_required:
+                cls.refresh_packages_list()
```

Comment:
Related to my comment above, I'm not sure refreshing the packages list explicitly is necessary here.

Reply:
Right! Also added a comment about it.